src/reverse_complement.rs

```diff
@@ -136,19 +153,32 @@ fn reverse_complement(seq: &mut [u8], tables: &Tables) {
     let div = len / 4;
     let rem = len % 4;
     unsafe {
-        let mut left = seq.as_mut_ptr() as *mut u16;
+        let p = seq.as_mut_ptr();
+        let xs = slice::from_raw_parts_mut(p as *mut u16, div);
         // This is slow if len % 2 != 0 but still faster than bytewise operations.
-        let mut right = seq.as_mut_ptr().offset(len as isize - 2) as *mut u16;
-        let end = left.offset(div as isize);
-        while left != end {
-            let tmp = tables.cpl16(*left);
-            *left = tables.cpl16(*right);
-            *right = tmp;
-            left = left.offset(1);
-            right = right.offset(-1);
-        }
+        let q = p.offset((div * 2 + rem) as isize);
+        let ys = slice::from_raw_parts_mut(q as *mut u16, div);
+
+        let thread_count = num_cpus::get();
+        let chunk_size = (div + thread_count - 1) / thread_count;
+        crossbeam::scope(|scope| {
+            for (a, b) in double_chunk(chunk_size, xs, ys) {
+                scope.spawn(move || {
+                    let mut left = a.as_mut_ptr();
+                    let mut right = b.as_mut_ptr().offset(b.len() as isize - 1);
+                    let end = left.offset(a.len() as isize);
+                    while left != end {
+                        let tmp = tables.cpl16(*left);
+                        *left = tables.cpl16(*right);
+                        *right = tmp;
+                        left = left.offset(1);
+                        right = right.offset(-1);
+                    }
+                });
+            }
+        });
```

Comment:
Can't we rewrite a lot of that without unsafe, limiting unsafe to the `&mut [u16]` creations?